tiflash: add descriptions for some conf items under [flash] section (#14880)
pingcap/docs: tiflash/tiflash-configuration.md

    tidb_status_addr = TiDB status port and address. # Multiple addresses are separated with commas.
    service_addr = The listening address of TiFlash Raft services and coprocessor services.

>     ## Introduced in v7.4.0, when the gap between the applied_index advanced by the current Raft state machine and the applied_index at the last disk flush exceeds compact_log_min_gap, TiFlash will execute the CompactLog command from TiKV and persist the data. Increasing this gap may reduce the disk flush frequency of TiFlash, thus reducing read latency in random write scenarios, but it will also increase memory overhead. Decreasing this gap may increase the disk flush frequency of TiFlash, thus alleviating memory pressure in TiFlash. However, at this stage, the disk flush frequency of TiFlash will not be higher than that of TiKV, even if this gap is set to 0.

Reviewer: ```suggestion
    ## Introduced in v7.4.0. When the gap between the `applied_index` advanced by the current Raft state machine and the `applied_index` at the last disk spilling exceeds `compact_log_min_gap`, TiFlash executes the `CompactLog` command from TiKV and spills data to disk. Increasing this gap might reduce the disk spilling frequency of TiFlash, thus reducing read latency in random write scenarios, but it might also increase memory overhead. Decreasing this gap might increase the disk spilling frequency of TiFlash, thus alleviating memory pressure in TiFlash. However, at this stage, the disk spilling frequency of TiFlash will not be higher than that of TiKV, even if this gap is set to 0.
```

Reviewer: TiFlash 落盘的英文可以参考这篇文档 https://docs.pingcap.com/tidb/dev/tiflash-spill-disk,

    ## It is recommended to keep the default value.
    # compact_log_min_gap = 200

>     ## Introduced in v5.0, when the number or size of the Region rows cached by TiFlash exceeds either of the following thresholds, TiFlash will execute the CompactLog command from TiKV and perform disk flushing.

Reviewer: ```suggestion
    ## Introduced in v5.0. When the number or size of the Region rows cached by TiFlash exceeds either of the following thresholds, TiFlash executes the `CompactLog` command from TiKV and spills data to disk.
```
